### PR TITLE
Add billing account name to the results of listing billing projects

### PR DESCRIPTION
**Pull request by breilly2**
Fixes: https://github.com/databiosphere/saturn-ui/issues/1275

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by breilly2** [comment]
Log shows:
`ERROR: gcloud crashed (TokenRevokeError): invalid_request`
Jenkins retest

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/UserApiServiceSpec.scala [hunk 1]
```diff
@@ -355,7 +355,7 @@ class UserApiServiceSpec extends ApiServiceSpec {
           assertResult(StatusCodes.OK) {
             status
           }
-          assertResult(List(RawlsBillingProjectMembership(project1.projectName, ProjectRoles.Owner, CreationStatuses.Error, Option(s"[Failure enabling api ${billingProjectMonitor.projectTemplate.services(1)}: this failed]")))) {
+          assertResult(List(RawlsBillingProjectMembership(project1.projectName, ProjectRoles.Owner, CreationStatuses.Error, Option(s"[Failure enabling api ${billingProjectMonitor.projectTemplate.services(1)}: this failed]"), Option(services.gcsDAO.accessibleBillingAccountName)))) {
```
Comment (davidangb) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/UserApiServiceSpec.scala:358]: this PR adds business logic of "if caller is not `ProjectRoles.Owner`,  do not return billing account name". Could you  add a test for that?

**Review by davidangb — CHANGES_REQUESTED** [review]
*  syntax looks good.
*  one request for a new test inline.
* please update swagger to reflect the response  payload change!
* please get Cloud Accounts approval for changing the API payload  (I recognize it is purely additive)

**Comment by dvoet** [comment]
the billing account stored in the database is not necessarily accurate. It is the billing account used when the project is created but it can be updated on the google side and the database will not reflect that.

**Comment by breilly2** [comment]
Thanks you @davidangb and @dvoet for your review and feedback. For the moment, I'm leaving this open but added an on-hold label while we investigate other ways to accomplish this. I will circle back later and either apply needed changes or close without merging once we get it all figure out.

**Comment by davidangb** [comment]
@breilly2 can this PR be closed?

**Comment by breilly2** [comment]
I'm going to say yes. Thanks for the reminder.

**Pull request closed without merging**